src/test/java/seedu/address/model/leave/LeaveTest.java

```diff
@@ -0,0 +1,128 @@
+package seedu.address.model.leave;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalLeaves.ALICE_LEAVE;
+import static seedu.address.testutil.TypicalLeaves.BOB_LEAVE;
+import static seedu.address.testutil.TypicalPersons.ALICE;
+import static seedu.address.testutil.TypicalPersons.BOB;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.leave.exceptions.EndBeforeStartException;
+import seedu.address.testutil.LeaveBuilder;
+
+public class LeaveTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Leave(null, null, null, null));
+        assertThrows(NullPointerException.class, () -> new Leave(null, null, null, null, null));
+    }
+
+    @Test
+    public void constructor_invalidLeave_throwsEndBeforeStartException() {
+        assertThrows(EndBeforeStartException.class, () -> new Leave(ALICE, "Alice's Maternity Leave",
+                ALICE_LEAVE.getEnd(), ALICE_LEAVE.getStart()));
+        assertThrows(EndBeforeStartException.class, () -> new Leave(BOB, "Bob's Paternity Leave",
+                BOB_LEAVE.getEnd(), BOB_LEAVE.getStart(), "Bob's Paternity Leave Description"));
```

Comment:
Can I confirm what the expected behaviour should be if start date == end date?

Reply:
I think it is best to implement a separate field class LeaveDate for start and end, then use a static method to better encapsulate the comparison class. What are your thoughts?